mr/src/test/java/org/apache/iceberg/mr/TestIcebergInputFormats.java

```diff
@@ -381,6 +386,46 @@ public void testCustomCatalog() throws IOException {
     testInputFormat.create(builder.conf()).validate(expectedRecords);
   }
 
+  @TestTemplate
+  public void testWorkerPool() throws Exception {
+    // 1.The ugi in the same thread will not change
+    final ExecutorService workerPool1 = ThreadPools.newWorkerPool("iceberg-plan-worker-pool", 1);
+    UserGroupInformation user1 =
+        UserGroupInformation.createUserForTesting("user1", new String[] {});
+    UserGroupInformation user2 =
+        UserGroupInformation.createUserForTesting("user2", new String[] {});
+    assertThat(setAndGetUgi(user1, workerPool1)).isEqualTo("user1");
+    assertThat(setAndGetUgi(user2, workerPool1)).isEqualTo("user1");
+    workerPool1.shutdown();
+
+    // 2.The ugi in different threads will be different
+    final ExecutorService workerPool2 = ThreadPools.newWorkerPool("iceberg-plan-worker-pool", 1);
+    assertThat(setAndGetUgi(user2, workerPool2)).isEqualTo("user2");
+    workerPool2.shutdown();
+  }
+
+  private String setAndGetUgi(UserGroupInformation ugi, ExecutorService workpool)
```

Comment:
i don't see any benefit in such a test, it doesn't test the production code. can't we make a method package private or change accessor and then use argument captor for the scan and validate that provided pool was set?

Reply:
Hi,thank you very much for your guidance!
Now I execute private methods through reflection(Learn from other unit tests and apply force).
Pass in workerpool for scanning verification and obtain ugi from the workerpool.

Reply:
Sorry for bothering you.
May I ask if you could review my code again when you have some free time?
I would really appreciate it.

Reply:
@lurnagao-dahua, the test looks good to me, thanks for adding it!   
Unfortunately, I don't have committer rights in the iceberg project, but I'll add my +1